app/bundles/EmailBundle/Helper/MailHelper.php

```diff
@@ -1506,6 +1518,13 @@ public function getCustomHeaders()
     {
         $headers = array_merge($this->headers, $this->getSystemHeaders());
 
+        // Personal and transactional emails do not contain unsubscribe header
+        $email                = $this->getEmail();
+        $addUnsubscribeHeader = $this->factory->getParameter('add_mass_mailing_header');
+        if (empty($email) || ('transactional' === $this->getEmailType() && !$addUnsubscribeHeader)) {
```

Comment:
If we know Email entity in the step, then we do not need store emailType. Do I miss something?

Reply:
EmailType is not stored in the Email entity. It is taken from the campaign.